### PR TITLE
SideBar and Modal implementation using useContext hook complete.

### PR DESCRIPTION
**Pull request by NStro03**
SideBar and Modal implementation using useContext hook complete.

### CODE REVIEW

File: src/context.js [hunk 1]
```diff
@@ -1 +1,38 @@
...
+        closeSideBar,
+        openModal,
+        closeModal
+    }
+
+    return <AppContext.Provider value={appValues}>{children}</AppContext.Provider>
+}
+// Can also wxport the below function to reduce the redundant code for accessing the useContext object.
+// export const useGlobalContext = () => {
+//     return useContext(AppContext);
```
Comment (Ishaan-flyfin) [inline src/context.js:37]: using this will be beneficial, from the next assignment onwards we can use this

File: src/context.js [hunk 2]
```diff
@@ -1 +1,38 @@
 import React, { useState, useContext } from 'react'
+export const AppContext = React.createContext(null);
```
Comment (Ishaan-flyfin) [inline src/context.js:2]: can you explain a little bit on the default value of `createContext`
not in the code, just here in the comment

File: src/pages/Home.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import React, { useContext } from 'react'
+import { FaBars } from 'react-icons/fa'
+import { AppContext } from '../context';
+
+const Home = () => {
+  const appData = useContext(AppContext);
+  console.log(appData.isSideBarOpen);
+  return (
+    <main>
+      <button onClick={() => appData.openSideBar()} className='sidebar-toggle'>
```
Comment (Ishaan-flyfin) [inline src/pages/Home.js]: as discussed earlier today, avoid using anonymous functions
`onClick={() => appData.openSideBar()}` 
can be simply written as
`onClick={appData.openSideBar}`

also check in other places for the same

File: src/Sidebar.js [hunk 4]
```diff
@@ -1,10 +1,47 @@
...
-  return <h2>sidebar</h2>
+  const sidebarLinks = links.map((link) => {
+    return (
+      <li key={link.id}>
+        <Link to={link.url}>{link.text}</Link>
+      </li>
+    )
+  })
+
+  const { isSideBarOpen, closeSideBar } = useContext(AppContext);
```
Comment (Ishaan-flyfin) [inline src/Sidebar.js]: try to declare variables, constants at the top of the component for better readability

**Review by Ishaan-flyfin — CHANGES_REQUESTED** [review]
LGTM!
just small comments are there

File: src/context.js (hunk 1 above)
Comment (NStro03) [reply src/context.js:37]: Sure will use this from next time.

File: src/context.js (hunk 2 above)
Comment (NStro03) [reply src/context.js:2]: This is what I got from react docs while investigating:
(https://reactjs.org/docs/context.html#contextprovider)
The defaultValue argument is only used when a component does not have a matching Provider above it in the tree. This default value can be helpful for testing components in isolation without wrapping them. Note: passing undefined as a Provider value does not cause consuming components to use defaultValue.

File: src/Sidebar.js (hunk 4 above)
Comment (NStro03) [reply src/Sidebar.js]: Sure, will take care of that.

File: src/pages/Home.js (hunk 3 above)
Comment (NStro03) [reply src/pages/Home.js]: Done

**Review by Ishaan-flyfin — APPROVED** [review]

**Pull request merged**